File: fixtures/sample.rb
require 'json'
require 'net/http'
require_relative 'config'
require_relative 'lib/helpers'

module Cacheable
  def cache_key
    "#{self.class.name}:#{id}"
  end

  def cached?
    !cache_key.nil?
  end
end

class DataProcessor
  include Comparable
  include Enumerable

  attr_accessor :name, :data
  attr_reader :status

  def initialize(name, data = [])
    @name = name
    @data = data
    @status = :pending
  end

  def process
    @data.reject(&:empty?).map(&:upcase)
  end

  def <=>(other)
    name <=> other.name
  end

  def each(&block)
    @data.each(&block)
  end

  private

  def validate
    raise ArgumentError, "Name required" if @name.nil?
  end
end

class ProcessConfig
  attr_accessor :max_retries, :timeout, :debug

  def initialize(max_retries: 3, timeout: 30, debug: false)
    @max_retries = max_retries
    @timeout = timeout
    @debug = debug
  end
end

def transform(input)
  input.map { |item| item.to_s.strip }.compact
end

def _internal_helper
  "private"
end

File: fixtures/sample.php
<?php

namespace App\Controllers;

use App\Models\User;
use App\Services\AuthService;
use Illuminate\Http\Request;
use Illuminate\Support\Facades\{Cache, Log};

require_once 'vendor/autoload.php';
include './helpers.php';

const MAX_RETRIES = 3;
const API_VERSION = "2.0";

interface Repository
{
    public function find(int $id): ?object;
    public function save(object $entity): bool;
    public function delete(int $id): bool;
}

trait Cacheable
{
    public function cacheKey(): string
    {
        return static::class . ':' . $this->id;
    }

    public function clearCache(): void
    {
        Cache::forget($this->cacheKey());
    }
}

trait Loggable
{
    public function logAction(string $action): void
    {
        Log::info("{$action} on " . static::class);
    }
}

class UserController
{
    use Cacheable;
    use Loggable;

    public function index(Request $request): array
    {
        return User::all()->toArray();
    }

    public function show(int $id): ?User
    {
        return User::find($id);
    }

    public function store(Request $request): User
    {
        $data = $this->validateInput($request);
        return User::create($data);
    }

    public static function create(): self
    {
        return new self();
    }

    private function validateInput(Request $request): array
    {
        return $request->validate([
            'name' => 'required|string',
            'email' => 'required|email',
        ]);
    }

    protected function authorize(User $user): bool
    {
        return $user->isAdmin();
    }
}

enum Status
{
    case Active;
    case Inactive;
    case Pending;
}

class ProcessConfig
{
    public int $timeout = 30;
    public int $retries = 3;
    private string $internalKey = 'secret';

    public function isValid(): bool
    {
        return $this->timeout > 0 && $this->retries > 0;
    }
}

function transform(array $data): array
{
    return array_map(fn($item) => strtoupper($item), $data);
}

function processQueue(string $queue): void
{
    // Process items from the named queue
    while ($item = array_shift($queue)) {
        handleItem($item);
    }
}
